### ISSUE TITLE
add a lint for useless transmutes

### ISSUE THREAD

**Issue by cmr** [rust-lang/rust#11400]
```rust
fn main() {
    unsafe { std::cast::transmute::<int, int> ( 52 ) };
}
```

Should have a warning. In general, `transmute::<T, T>` should be warned against.

**Comment by tshepang** [comment]
That sample code should now be:

```
fn main() {
    unsafe { std::mem::transmute::<int, int> ( 52 ) };
}
```

**Comment by Manishearth** [comment]
Filed https://github.com/Manishearth/rust-clippy/issues/441
